terra_notebook_utils/drs.py

```diff
@@ -121,6 +125,25 @@ def copy_to_local(drs_url: str,
         logger.info(f"Downloading {drs_url} to {filepath}")
         blob.download_to_file(fh)
 
+def head(drs_url: str,
+         n: int = 1,
+         workspace_name: Optional[str]=WORKSPACE_NAME,
+         google_billing_project: Optional[str]=WORKSPACE_GOOGLE_PROJECT):
+    """
+    Check access to a DRS object by attempting to access its first byte of data.
+    """
+    assert drs_url.startswith("drs://")
+    enable_requester_pays(workspace_name, google_billing_project)
+    client, info = resolve_drs_for_gs_storage(drs_url)
+    blob = client.bucket(info.bucket_name, user_project=google_billing_project).blob(info.key)
+    try:
+        # don't expand compressed files, to save time
+        return blob.download_as_bytes(start=0, end=n, raw_download=True)
```

Comment:
Consider dropping `raw_download`:
  - As far as I'm aware, there are no BDCat files that meet the requirements for decompressive transcoding
  - If there are such files, returning decompressed bytes will be more in line with user expectations
  - Decompressing the first few bytes of a file is not a large time penalty

See documentation for Google decompressive transcoding:
https://cloud.google.com/storage/docs/transcoding